src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfoOfT.WriteHelpers.cs

```diff
@@ -267,7 +266,11 @@ public partial class JsonTypeInfo<T>
                             }
                             else
                             {
-                                await utf8Json.FlushAsync(cancellationToken).ConfigureAwait(false);
+                                FlushResult result = await utf8Json.FlushAsync(cancellationToken).ConfigureAwait(false);
+                                if (result.IsCanceled || result.IsCompleted)
+                                {
+                                    break;
```

Comment:
Is this the right behavior? If the flush is canceled or pipe is completed we just stop writing? From a user perspective they'll "successfully" have written an object which seems a bit odd.